plugins/GSdx/GSRendererDX.cpp

```diff
@@ -147,6 +147,193 @@ void GSRendererDX::EmulateZbuffer()
 	}
 }
 
+void GSRendererDX::EmulateTextureShuffleAndFbmask()
+{
+	size_t count = m_vertex.next;
+	GSVertex* v = &m_vertex.buff[0];
+
+	bool m_IS_DX1011 = false;
```

Comment:
No, you need to set (cache) the value during the init of the object.

Otherwise a much much nicer implementation would be to create a virtual function with 2 implementations. 1 for Dx9 and 1 for Dx10. You can look at SetupIA if you want an example.